crates/agentgateway/src/client/mod.rs

```diff
@@ -302,6 +323,27 @@ impl Connector {
 					buf: Default::default(),
 				};
 				let mut socket = Socket::from_hbone(Arc::new(stream::Extension::new()), pool_key.dst, rw);
+				let connect_ms = connect_start.elapsed().as_millis();
+				if let Some(m) = &self.metrics {
+					let labels = crate::telemetry::metrics::ConnectLabels {
+						transport: agent_core::strng::RichStrng::from(transport_name).into(),
+					};
+					m.upstream_connect_duration
+						.get_or_create(&labels)
+						.observe((connect_ms as f64) / 1000.0);
```

Comment:
nit: do we want to use micros and 1_000_000? seems the same but more precise?

Reply:
Yeah can add a comment here to make it less confusing!